src/tls13.c

```diff
@@ -8458,24 +8467,34 @@ int wolfSSL_accept_TLSv13(WOLFSSL* ssl)
 #ifndef NO_CERTS
     /* allow no private key if using PK callbacks and CB is set */
     if (!havePSK) {
-        if (!ssl->buffers.certificate ||
-            !ssl->buffers.certificate->buffer) {
-
-            WOLFSSL_MSG("accept error: server cert required");
-            WOLFSSL_ERROR(ssl->error = NO_PRIVATE_KEY);
-            return WOLFSSL_FATAL_ERROR;
-        }
-
-    #ifdef HAVE_PK_CALLBACKS
-        if (wolfSSL_CTX_IsPrivatePkSet(ssl->ctx)) {
-            WOLFSSL_MSG("Using PK for server private key");
+    #if defined(OPENSSL_ALL) || defined(OPENSSL_EXTRA) || \
+        defined(WOLFSSL_NGINX) || defined (WOLFSSL_HAPROXY)
+        if (ssl->ctx->certSetupCb != NULL) {
+            WOLFSSL_MSG("CertSetupCb set. server cert and "
+                        "key not checked");
         }
         else
     #endif
-        if (!ssl->buffers.key || !ssl->buffers.key->buffer) {
-            WOLFSSL_MSG("accept error: server key required");
-            WOLFSSL_ERROR(ssl->error = NO_PRIVATE_KEY);
-            return WOLFSSL_FATAL_ERROR;
+        {
+            if (!ssl->buffers.certificate ||
+                !ssl->buffers.certificate->buffer) {
+
+                WOLFSSL_MSG("accept error: server cert required");
+                WOLFSSL_ERROR(ssl->error = NO_PRIVATE_KEY);
```

Comment:
WOLFSSL_ERROR as a macro may map to nothing and this assignment won't happen.
Separate out assignment from macro call.

Reply:
Separated out.